src/physics/nonlinear_solid.hpp

```diff
@@ -233,14 +224,92 @@ class NonlinearSolid : public BasePhysics {
   std::unique_ptr<mfem::ParGridFunction> deformed_nodes_;
 
   /**
-   * @brief Solve the Quasi-static operator
+   * @brief Mass matrix
+   */
+  std::unique_ptr<mfem::HypreParMatrix> M_mat_;
+
+  /**
+   * @brief Damping matrix
+   */
+  std::unique_ptr<mfem::HypreParMatrix> C_mat_;
+
+  /**
+   * @brief Jacobian (or "effective mass") matrix
+   */
+  std::unique_ptr<mfem::HypreParMatrix> J_mat_;
+
+  /**
+   * @brief Mass bilinear form object
+   */
+  std::unique_ptr<mfem::ParBilinearForm> M_;
+
+  /**
+   * @brief Damping bilinear form object
+   */
+  std::unique_ptr<mfem::ParBilinearForm> C_;
+
+  /**
+   * @brief Stiffness bilinear form object
+   */
+  std::unique_ptr<mfem::ParNonlinearForm> H_;
+
+  /**
+   * @brief zero vector of the appropriate dimensions
    */
-  virtual void quasiStaticSolve();
+  mfem::Vector zero_;
 
   /**
    * @brief Nonlinear system solver instance
    */
   EquationSolver nonlin_solver_;
+
+  /**
+   * @brief mfem::Operator for computing the weighted residual
+   */
+  StdFunctionOperator residual_;
+
+  /**
+   * @brief the system of ordinary differential equations for the physics module
+   */
+  SecondOrderODE ode2_;
+
+  /**
+   * @brief alias for the reference mesh coordinates
+   *
+   *   this is used to correct for the fact that mfem's hyperelastic
+   *   material model is based on the nodal positions rather than nodal displacements
+   */
+  mfem::Vector x_;
+
+  /**
+   * @brief used to communicate the ODE solver's predicted displacement to the residual operator
+   */
+  mfem::Vector u_;
+
+  /**
+   * @brief used to communicate the ODE solver's predicted velocity to the residual operator
+   */
+  mfem::Vector du_dt_;
+
+  /**
+   * @brief the previous acceleration, used as a starting guess for newton's method
+   */
+  mfem::Vector previous_;
+
+  // TODO: wrap mfem's second order ODE solvers to incorporate this boundary condition info
+  //
+  // temporary values used to compute finite difference approximations
+  // to the derivatives of constrained degrees of freedom
+  mfem::Vector U_minus_;
+  mfem::Vector U_;
```

Comment:
I'm still worried about collisions with `u_`. Maybe `u_temp_`?

Reply:
These members also still need doxygen-style comments.

Reply:
After this PR is merged, these should go away as the MFEM second order ODE solver will be wrapped.